app/message_error_handler.py

```diff
@@ -0,0 +1,107 @@
+        'exceptionClass': exception_class,
+        'exceptionMessage': exception_trace,
+    }
+
+    response = requests.post(f'{Config.EXCEPTION_MANAGER_URL}/reportexception', json=exception_report_payload)
+    response.raise_for_status()
+    return response.json()
+
+
+def _quarantine_message(body: bytes, message_hash, exception_class, headers):
+    quarantine_headers = headers or {}
+    quarantine_headers['quarantineOrigin'] = {
+        'exchange': Config.RABBIT_EXCHANGE,
+        'queue': Config.RABBIT_QUEUE,
+        'routing-keys': [Config.RABBIT_ROUTING_KEY]
+    }
+    _quarantine_message_in_exception_manager(body, message_hash, exception_class, quarantine_headers)
+    _quarantine_message_in_rabbit(body, quarantine_headers)
+
+
+def _quarantine_message_in_exception_manager(body: bytes, message_hash, exception_class, headers):
+    quarantine_payload = {
+        'messageHash': message_hash,
+        'messagePayload': base64.b64encode(body).decode(),
+        'service': Config.NAME,
+        'queue': Config.RABBIT_QUEUE,
+        'exceptionClass': exception_class,
+        'routingKey': Config.RABBIT_ROUTING_KEY,
+        'contentType': 'application/json',
+        'headers': headers,
+    }
+
+    response = requests.post(f'{Config.EXCEPTION_MANAGER_URL}/storeskippedmessage', json=quarantine_payload)
+    response.raise_for_status()
+
+
+def _quarantine_message_in_rabbit(body: bytes, headers):
+    properties = BasicProperties(content_type='application/json', headers=headers)
+    with RabbitContext(queue_name=Config.RABBIT_QUARANTINE_QUEUE) as rabbit:
+        rabbit.channel.basic_publish(Config.RABBIT_QUARANTINE_EXCHANGE,
+                                     rabbit.queue_name,
+                                     body,
+                                     properties=properties,
+                                     mandatory=True)
+
+
+def _peek_message(message_hash, body: bytes):
+    peek_payload = {
+        'messageHash': message_hash,
+        'messagePayload': base64.b64encode(body).decode(),
+    }
+
+    response = requests.post(f'{Config.EXCEPTION_MANAGER_URL}/peekreply', json=peek_payload)
+    response.raise_for_status()
+
+
+def handle_message_error(message_body: bytes, exception: Exception, channel, delivery_tag, headers):
+    message_hash = hashlib.sha256(message_body).hexdigest()
+    exception_class = type(exception).__name__
+
+    try:
+        advice = _report_exception_for_advice(message_hash, Config.NAME, Config.RABBIT_QUEUE, exception_class,
+                                              repr(exception))
+
+        if advice.get('skipIt'):
+            logger.warn('Attempting to quarantine and ack bad message message', message_hash=message_hash,
+                        queue=Config.RABBIT_QUEUE, routing_key=Config.RABBIT_ROUTING_KEY)
+            _quarantine_message(message_body, message_hash, exception_class, headers)
+            channel.basic_nack(delivery_tag=delivery_tag, requeue=False)
```

Comment:
I would add a log line here to say that the message has been quarantined successfully